pkg/kubelet/types/pod_update.go

```diff
@@ -147,6 +147,25 @@ func IsCriticalPod(pod *v1.Pod) bool {
 	return IsCritical(pod.Namespace, pod.Annotations) || (pod.Spec.Priority != nil && IsCriticalPodBasedOnPriority(*pod.Spec.Priority))
 }
 
+// Preemptable returns true if preemptor pod can preempt preemptee pod:
+//   - If preemptor is nil
+//       - If preemptee's priority is greater than or equal to SystemCriticalPriority,
+//         preemptee can not be preempted (return false)
+//       - If preemptee bears the critical pod annotation key, it can not be preempted (return false)
+//   - If preemptor's is greater than preemptee's priority, it's preemptable (return true)
+//   - Otherwise, can not preempt (return false)
+func Preemptable(preemptor, preemptee *v1.Pod) bool {
```

Comment:
Is it implicit that the preemptor is a critical pod?  Maybe add the check since this is a general until function.

Reply:
I think it's not necessary: if priority/preemption is enabled, we just compare priority; otherwise, we check preemptee's critical annotation :).